echopype/consolidate/loc_utils.py

```diff
@@ -0,0 +1,162 @@
+            output_message = (
+                "Coordinate variables contain zero(s). "
+                "Interpolation may be negatively impacted, "
+                "consider handling these values before calling ``add_location``."
+            )
+
+        if echodata.sonar_model.startswith("EK"):
+            # Initialize good datagram type list
+            good_datagram_types = []
+
+            # Iterate through potential datagram type sets
+            potential_datagram_type_set = set([None, "MRU1", "IDX"])
+            potential_datagram_type_set.discard(datagram_type)
+            for potential_datagram_type in potential_datagram_type_set:
+                # Grab variables from EchoData object
+                if potential_datagram_type is not None:
+                    potential_lat_var = echodata["Platform"].get(
+                        f"latitude_{potential_datagram_type.lower()}"
+                    )
+                    potential_lon_var = echodata["Platform"].get(
+                        f"latitude_{potential_datagram_type.lower()}"
+                    )
+                elif potential_datagram_type is None:
+                    potential_lat_var = echodata["Platform"].get("latitude")
+                    potential_lon_var = echodata["Platform"].get("longitude")
+
+                # Check if loc vars are invalid
+                if validity_check in ["missing", "all_nan"]:
+                    potential_invalid_bool = compute_invalid_check(
+                        potential_lat_var, potential_lon_var, "missing"
+                    ) or compute_invalid_check(potential_lat_var, potential_lon_var, "all_nan")
+                elif validity_check in ["some_nan", "some_zero"]:
+                    potential_invalid_bool = compute_invalid_check(
+                        potential_lat_var, potential_lon_var, "some_nan"
+                    ) or compute_invalid_check(potential_lat_var, potential_lon_var, "some_zero")
+                # Append to good datagram types
+                if not potential_invalid_bool:
+                    good_datagram_types.append(potential_datagram_type)
+
+            if len(good_datagram_types) > 0:
+                # Append to output message
+                output_message = (
+                    output_message
+                    + f" Consider setting datagram_type to any of {good_datagram_types}."
+                )
+
+        # Raise error / warning depending on validity_check value
+        if validity_check in ["missing", "all_nan"]:
+            raise ValueError(output_message)
+        elif validity_check in ["some_nan", "some_zero"]:
+            logger.warning(output_message)
+
+
+def check_loc_time_dim_duplicates(echodata: EchoData, time_dim_name: str) -> None:
+    """Check if there are duplicates in time_dim_name"""
+    if len(np.unique(echodata["Platform"][time_dim_name].data)) != len(
+        echodata["Platform"][time_dim_name].data
+    ):
```

Comment:
I wonder how expensive `unique` is. We can keep this in place for now. An alternative I can think of is to check if any of `diff(time_dim_name) == 0`. But of course this would only work if the duplicates happen in consecutive indices (which is most of the cases, but hard to say if all cases). 

How about we give user the option to skip this check in `add_location` and accept that it may fail?

Reply:
I see what you mean. This could be very expensive since it will require everything to be loaded into memory. This would be pretty bad in the case where we have many many Echodata objects concatenated together, calibrated, and add_depth is used on that. But I'm not quite sure if this should be an option. It feels just a bit awkward and too specific.

Reply:
Maybe this won't be too bad though. I didn't encounter any memory issues on this specific part of adding depth when I was running this Krill NASC GPS notebook: https://github.com/OSOceanAcoustics/echopype-examples/pull/52 and I was working with Sv from 170 RAW files.

Reply:
Hmm interesting, thanks for trying it out. Maybe once we get to a point to memory profile all functions in the pipeline together we can see how bad that it.